src/components/Forms/Select.js

```diff
@@ -0,0 +1,351 @@
+    hintText           : PropTypes.string,
+    /** Uniq id for Select */
+    id                 : PropTypes.string,
+    /** Handled by FormComponent after running built in validations */
+    isValid            : PropTypes.bool,
+    /** onOpen callback */
+    onOpen             : PropTypes.func,
+    /** onClose callback */
+    onClose            : PropTypes.func,
+    /** onFocus callback */
+    onFocus            : PropTypes.func,
+    /** onSelect callback returns option object {label: , value:}*/
+    onSelect           : PropTypes.func,
+    /** onBlur callback */
+    onBlur             : PropTypes.func,
+    /** Mark the field as required.  */
+    required           : PropTypes.bool,
+    /** Control the component selection manually */
+    selectedOption     : PropTypes.shape({
+      label: PropTypes.string,
+      value: PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.number,
+        PropTypes.bool
+      ])
+    }),
+    /** Error from server to show ServerError message */
+    serverError        : PropTypes.string,
+    /** Wrapper styles, mainly used for custom width */
+    style              : PropTypes.object,
+    /** Text to show for validation error  */
+    validationErrorText: PropTypes.string,
+    /** Value will make TextField a controlled component  */
+  }
+
+  static defaultProps = {
+    disabled    : false,
+    defaultOption: null,
+    onFocus: () => {}, // eslint-disable-line no-empty-function
+    onBlur: () => {}, // eslint-disable-line no-empty-function
+    onOpen: () => {}, // eslint-disable-line no-empty-function
+    onClose: () => {}, // eslint-disable-line no-empty-function
+    onSelect: () => {} // eslint-disable-line no-empty-function
+  }
+
+  state = {
+    isOpen: false,
+    selectedOption: this.props.defaultOption || this.props.selectedOption || null
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { selectedOption } = nextProps
+    if (typeof selectedOption !== 'undefined' && selectedOption !== this.state.selectedOption) {
+      this.setState({selectedOption: selectedOption})
+    }
+  }
+
+  getValue = () => {
+    return (this.state.selectedOption && this.state.selectedOption.value) || null
+  }
+
+  _handleClose = () => {
+    this.trigger.focus()
+    this.props.onClose()
+  }
+
+  _handleRequestChange = (open) => {
+    if (!this.props.disabled) {
+      this.setState({isOpen: open})
+    }
+  }
+
+  _handleFocus = (e) => {
+    this.setState({isFocused: true}, () => {
+      this.props.onFocus(e)
+    })
+  }
+
+  _handleBlur = (e) => {
+    this.setState({isFocused: false}, () => {
+      this.props.onBlur(e)
+    })
+  }
+
+  _handleSelect = (option) => {
+    const { onSelect, selectedOption } = this.props
+
+    // For manual control
+    if (!selectedOption) {
+      this.setState({selectedOption: option}, () => {
+        onSelect(option)
+      })
+    } else {
+      onSelect(option)
```

Comment:
I made a comment below, but we discussed trying to standardize callback signatures to take the event and props/state to allow the parent components of `Snack` components to easily predict how callbacks will be used.